### PR TITLE
finish except matchBytes

### PR DESCRIPTION
**Pull request by Bert-Z**
nearly finish rule

### CODE REVIEW

**Comment by Bert-Z** [comment]
boost:regex have the bytes regex_match, I may try it later

**Comment by Bert-Z** [comment]
finish writing

File: src/ptrace/rule.cpp [hunk 1]
```diff
@@ -1,49 +1,110 @@
...
+RuleImpl::RuleImpl(int ID, int target_syscall, const std::string &name, RuleLevel level)
+    : ID(ID), target_syscall(target_syscall), name(name), level(level){};
 
-RuleCheckMsg RuleImpl::check(const core::SyscallParameter & sp) {
-    return RuleCheckMsg();
+RuleCheckMsg RuleImpl::check(const core::SyscallParameter &sp)
+{
+    RuleCheckMsg rcm = {false, ID, ""};
+
+    for (auto f : rulevalues)
```
Comment (Veiasai) [inline src/ptrace/rule.cpp]: auto & f

File: src/ptrace/app/main.cpp [hunk 2]
```diff
@@ -33,14 +34,17 @@ void startChild(const char * target) {
...
+    if (argc != 4) {
+        cout << "please input the target file name and log level and configuration file name\n";
         return -1;
     }
 
     startChild(argv[1]);
     initLogger(argv[2]);
-    
+    // I comment it for passing the coding examination
+    // YAML::Node config=YAML::LoadFile(argv[3]);
```
Comment (Veiasai) [inline src/ptrace/app/main.cpp]: ???

**Pull request merged**